app/assets/javascripts/pages/data_portal/DataPortalReportPage.js

```diff
@@ -0,0 +1,153 @@
+        .always(this.render.bind(this));
+    },
+
+    /**
+     * Return the sorted list of visible indicators
+     * NOTE: the access indicators are always displayed first
+     * @returns {object[]} widgets
+     */
+    _getVisibleIndicators: function () {
+      return this.indicatorsCollection.toJSON().filter(function (indicator) {
+        return indicator.visible;
+      }).sort(function (a, b) {
+        var aIsAccess = a.category === App.Helper.Indicators.CATEGORIES.ACCESS;
+        var bIsAccess = b.category === App.Helper.Indicators.CATEGORIES.ACCESS;
+        if (aIsAccess && !bIsAccess) return -1;
+        if (!aIsAccess && bIsAccess) return 1;
+        return 0;
+      });
+    },
+
+    /**
+     * Create the widget containers
+     * @param {number} count - number of containers to create
+     * @returns {DocumentFragment} containers
+     */
+    _createWidgetsContainer: function (count) {
+      var fragment = new DocumentFragment();
+
+      for (var i = 0, j = count; i < j; i++) {
+        var div = document.createElement('div');
+        div.classList.add('grid-s-12', 'grid-l-6');
+        var widget = document.createElement('div');
+        widget.classList.add('c-widget');
+        div.appendChild(widget);
+        fragment.appendChild(div);
+      }
+
+      return fragment;
+    },
+
+    /**
+     * Update the size of the widget container depending on
+     * if the category of the indicator
+     * @param {object} indicator
+     * @param {object} widgetContainer
+     */
+    _updateWidgetContainer: function (indicator, widgetContainer) {
+      var visibleIndicators = this._getVisibleIndicators();
+      var index = _.findIndex(visibleIndicators, { id: indicator.id });
+      var indicator = visibleIndicators[index];
+      var isAccess = indicator.category === App.Helper.Indicators.CATEGORIES.ACCESS;
+
+      if (isAccess) {
+        widgetContainer.classList.remove('grid-l-6');
+      }
+    },
+
+    /**
+     * Render the widgets
+     */
+    _renderWidgets: function () {
+      if (this._loadingError) {
+        new App.View.RetryMessageView({
+          el: this.widgetsContainer,
+          label: 'Unable to load the indicators',
+          callback: this._fetchData.bind(this)
+        });
+        return;
+      }
+
+      // We create the containers for the widgets
+      var fragment = this._createWidgetsContainer(this.options.indicators.length);
+
+      this.widgetsContainer.innerHTML = ''; // We ensure it's empty first
+      this.widgetsContainer.appendChild(fragment);
+
+      // If the widget instances already exist, we remove the listeners
+      // and reset their pointer
+      if (this.widgets) {
+        this.widgets.forEach(function (widget) {
+          this.stopListening(widget);
+        }, this);
+        this.widgets = [];
+      } else {
+        this.widgets = [];
+      }
+
+      // We instantiate the widget views
+      this.options.indicators.forEach(function (indicator, index) {
```

Comment:
Good idea. What do you think about making the sort in this file for now and update the task about `_getVisibleIndicators` to include what you suggest?

I would do it later because the sort is also made in other files that have nothing to do with this PR.